src/relax/transform/gradient.cc

```diff
@@ -408,27 +408,28 @@ class GradientMutator : private ExprMutator {
 /*!
  * \brief This is the internal function of tvm::relax::transform::Gradient.
  * \param mod The module
- * \param gvar The GlobalVar of the specified function
+ * \param func_name The name of the specified function
  * \param require_grads The relax variables whose adjoints are needed.
  * \return The module after transformation.
  */
-IRModule Gradient(const IRModule& mod, const GlobalVar& gvar, Optional<Array<Var>> require_grads) {
-  auto* func = mod->Lookup(gvar).as<FunctionNode>();
-  CHECK(func) << "Relax function " << gvar->name_hint << " is not found";
+IRModule Gradient(const IRModule& mod, const String& func_name,
+                  Optional<Array<Var>> require_grads) {
+  auto* func = mod->Lookup(func_name).as<FunctionNode>();
```

Comment:
`Lookup` uses `GetGlobalVar`, which directly throws error if the global var of the given name does not exist.
https://github.com/mlc-ai/relax/blob/378d4a8b5f43434a8f9bd30cf2f1d21fdd2126ac/src/ir/module.cc#L286-L289
We can use `ContainGlobalVar` to check if the global var exists first.